src/modules/IsnsServer.rb

```diff
@@ -351,7 +351,9 @@ def isnsadm(params, ret_result = false)
 
     def isnsadm_control
       if !@isctrlnode
-        @isctrlnode = isnsadm("--register control")
+        res = isnsadm("--register control", ret_result = true)
+        @isctrlnode = res["exit"] == 0
```

Comment:
The referenced bug is a 2-in-1 bug.
1. yast dies if isnsadm returns an error
2. In some cases the isns DB is corrupt so that isnsadm returns an error.

This only fixes #1.
#2 is not our (yast-team's) business.